platforms/substrate/configuration/roles/create/genesis/tasks/check_vault.yaml

```diff
@@ -0,0 +1,20 @@
+##############################################################################################
+#  Copyright Accenture. All Rights Reserved.
+#
+#  SPDX-License-Identifier: Apache-2.0
+##############################################################################################
+
+# This task checks if the genesis is already present in the vault
+- name: Check if genesis is present in the Vault
+  environment:
+    VAULT_ADDR: "{{ vault.url }}"
+    VAULT_TOKEN: "{{ vault.root_token }}"
+  shell: |
+    vault kv get {{ vault.secret_path | default('secretsv2') }}/{{ component_ns }}/genesis
```

Comment:
Add key path like genesis otherwise this will succeed even when the version has been deleted